### ISSUE TITLE
Setting environment variables in production does not work

### ISSUE THREAD

**Issue by toddw** [TrueCar/gluestick#952]
# Bug
Environment variables should be definable at runtime. Currently, even with [gluestick-env-expose-ast](https://github.com/TrueCar/gluestick/tree/next/packages/gluestick-env-expose-ast) this does not appear to be the case.

## Reproduction Steps
_Copy paste friendly_
```
# 1. Create App
gluestick new gs-env-test

# 2. Enter Folder
cd gs-env-test

# 3. Install env plugin dependency
npm install gluestick-env-expose-ast

# 4. Update plugins file to include env plugin
echo 'export default ["gluestick-env-expose-ast"];' > src/gluestick.plugins.js

# 5. Update application config to use env variable for `name`
echo 'const config = {
  name: process.env.NAME,
  assetPath: process.env.ASSET_URL || "/assets",
  head: {},
  logger: {
    pretty: false,
    level: "warn"
  }
};
export default config;
' > src/config/application.js

# 6. Update Home component to use variable
echo 'import React from "react";
import config from "config/application";

export default function Home () {
  return (
    <div><h1>{config.name}</h1></div>
  );
}' > src/apps/main/components/Home.js

# 7. Build app with name set to Todd
NAME=Todd NODE_ENV=production gluestick build

# 8. Start app with name set to Zeke
NAME=Zeke NODE_ENV=production gluestick start -P
```

Open browser to [http://localhost:8880/](http://localhost:8880)

## Current Result
Even though the environment variable `NAME` was set to `Zeke` when the app was started it used `Todd` which was set during the build.

## Expected Result
The h1 tag should say `Zeke` since that is the new value that the app is started with. You should be able to set `NAME` to anything you want when the app starts and it should override whatever was set during build.

## Notes
In 0.10.x all of the environment variables detected in _src/config/application.js_ with the [detectEnvironmentVariables](https://github.com/TrueCar/gluestick/blob/develop/src/lib/detectEnvironmentVariables.js) script would be exposed globally via [window.\_\_GS_ENV_VARS\_\_](https://github.com/TrueCar/gluestick/blob/develop/src/lib/server/Body.js#L62). Then we wrote a babel plugin called [babel-plugin-gluestick](https://github.com/TrueCar/babel-plugin-gluestick) to detect references to `process.env.*` in _src/config/application.js_ and replace them with `window.__GS_ENV_VARS__.*` during the client build so that the environment variables are not hard coded into the build. The babel plugin was integrated into the webpack config file called [getWebpackClientConfig](https://github.com/TrueCar/gluestick/blob/develop/src/config/getWebpackClientConfig.js#L80-L96) Instead they would rely on the values set dynamically on the server when the app is started. This let us pre-build without worrying about what the environment variables would eventually be.

It is very important that we only expose the environment variables referenced in _src/config/application.js_ as exposing ALL environment variables would be a major security issue.

**Comment by toddw** [comment]
I appreciate that we are moving this functionality into a plugin. However, am I the only one who feels this should be the default GlueStick behavior?

**Comment by toddw** [comment]
This is fixed in 1.7.0 thanks to #953 
Need to include the options for the plugin
```
export default [
  {
    plugin: "gluestick-env-expose-ast",
    options: {
      exposeRuntime: true
    }
  }
];
```
